app/models/reference_repository.rb

```diff
@@ -52,6 +52,9 @@ def self.update_from_client(client)
     rr = ReferenceRepository.find_or_create_by(client_id: client.uid)
     if client.re3data_id && (not rr.re3doi)
       rr.re3doi = client.re3data_id
+      if !rr.validate
+        ReferenceRepository.find_by(re3doi: client.re3data_id, client_id: nil).try(:destroy)
```

Comment:
you might not need the if-clause if you put this line at the very beginning of the method. I think [.try method](https://apidock.com/rails/v3.2.1/Object/try) will just return a Nill if the object doesn't exist. And delete it if it exists.

Reply:
Thanks for taking a look at this.

I think that you are right.  The code should technically work without clause, but it does help explain why we are eventually running the destroy method.

I think will help "future us" understand this code when we look back on it in the future.